client/my-sites/navigation/index.jsx

```diff
@@ -26,7 +25,7 @@ class MySitesNavigation extends React.Component {
 					siteBasePath={ this.props.siteBasePath }
 					onClose={ this.preventPickerDefault }
 				/>
-				<Sidebar path={ this.props.path } siteBasePath={ this.props.siteBasePath } />
+				{ this.props.sidebar }
```

Comment:
I wonder what's the benefit of specifying the logic of which component to load here in the parent component, vs doing it here?

Reply:
I think the benefit is that we are keeping the component simpler. Not sure what other benefit there is.

Reply:
I think an extra layer of indirection is only complicating things. But happy to keep it as-is if you feel this is better.